subprojects/internal-testing/src/main/groovy/org/gradle/test/precondition/RequiresExtension.groovy

```diff
@@ -69,12 +69,19 @@ class RequiresExtension implements IAnnotationDrivenExtension<Requires> {
     /**
      * Common method for handling annotations.
      */
-    void visitAnnotation(Requires annotation, SpecElementInfo feature) {
-        def predicateClassNames = annotation.value() as List
+    void visitAnnotation(Requires annotation, SpecElementInfo specOfFeature) {
+        List<Class> predicateClassNames = annotation.value() as List<Class>
+
+        if (specOfFeature.parent instanceof SpecInfo) {
```

Comment:
🤔 I wonder if there is a more idiomatic way of gathering all the annotations for this feature from parents. @leonard84 Do you have a better way of gathering the annotations that led to this feature?